wip_pacta_data_qa.Rmd

```diff
@@ -0,0 +1,545 @@
+knitr::opts_chunk$set(echo = FALSE)
+library(tidyverse)
+library(DT)
+
+source("1_portfolio_check_initialisation.R")
+```
+
+# Use case
+This document can be used to perform Data QA on processed Portfolio Input files.
+The aim of this analysis is to ensure data consistency and find holdings, that may not be considered in the PACTA analysis yet, but could be included in one of the below described ways in order to increase portfolio coverage.
+
+## Read in data
+
+For the analysis to be run, the processed portfolio inputs as well as the company level results for both corporate bonds and listed equity need to be read in. Given the normal PACTA project structure is used, this .Rmd will automatically locate the corresponding files and read them in. In case the folder structure of the project was changed, the paths need to be adjusted manually (this should rarely be the case).
+
+```{r}
+# read in if report has been run successfully, but data is not currently in the workspace
+total_portfolio <- paste0(project_location,"/30_Processed_Inputs/",project_name,"_total_portfolio.rda")
+bonds_company <- paste0(project_location,"/40_Results/Meta Investor/Bonds_results_company.rda")
+equity_company <- paste0(project_location,"/40_Results/Meta Investor/Equity_results_company.rda")
+
+
+result_data_qa <- readRDS(total_portfolio)
+bonds_results_company <- readRDS(bonds_company)
+equity_results_company <- readRDS(equity_company)
+
+```
+
+## Overview
+### variables, data types, etc.
+Brief overview of variable names and classes of the portfolio input file
+```{r}
+result_data_qa %>% glimpse()
+```
+
+### Total holdings and value
+Overall number of holdings and aggregated value in USD of the portfolio input, by asset type
+```{r}
+total_overview <- result_data_qa %>% 
+  filter(investor_name != "Meta Investor") %>% 
+  group_by(asset_type) %>% 
+  summarise(total_holdings = n(),
+            sum_value_usd = round(sum(value_usd, na.rm = T))) %>% 
+  ungroup()
+datatable(total_overview)
+```
+
+
+
+### Total holdings and value per investor
+Number of holdings and aggregated value in USD of the portfolio input - grouped by investor.
+"Meta Investor" is the total aggregation across investors.
+```{r}
+investors_overview <- result_data_qa %>% 
+  group_by(investor_name) %>% 
+  summarise(total_holdings = n(), sum_value_usd = round(sum(value_usd, na.rm = T))) %>% 
+  ungroup()
+datatable(investors_overview)
+```
+
+### Total holdings and value per investor and portfolio
+Number of holdings and aggregated value in USD of the portfolio input - grouped by investor and portfolio
+
+* **total_holdings** is the number of holdings per investor and portfolio
+* **sum_value_usd** is the aggregated value of the holdings in USD by investor and portfolio
+* **sum_market_value** .... see comment below
+* **pct_holdings** is the share of holdings of a specific PF, relative to the overall number of holdings in the analysis. It's the weight of the PF in pure terms of different holdings.
```

Comment:
Weight of the portfolio in all of the overall investor's portfolios? I'm not sure I get this one and the next.